lib/internal/bootstrap/pre_execution.js

```diff
@@ -332,6 +335,61 @@ function initializeDeprecations() {
     enumerable: false,
     configurable: true
   });
+
+  let warnedForProto = false;
+  const ObjectPrototype = ObjectGetPrototypeOf({});
+  function warnOnProto() {
+    if (!warnedForProto) {
+      warnedForProto = true;
+      process.emitWarning('usage of Object.prototype.__* properties are a' +
+        ' common security concern, use static methods on Object instead');
+    }
+  }
+  const legacy = [
+    '__proto__',
+    '__defineGetter__',
+    '__defineSetter__',
+    '__lookupGetter__',
+    '__lookupSetter__',
+  ];
+  for (let i = 0; i < legacy.length; i++) {
+    const prop = legacy[i];
+    const protoDescriptor = ObjectGetOwnPropertyDescriptor(
+      ObjectPrototype,
+      prop);
+    if ('value' in protoDescriptor) {
```

Comment:
```suggestion
    if (protoDescriptor && 'value' in protoDescriptor) {
```
Fall through if the descriptor is not available.